content/en/docs/concepts/workloads/pods/ephemeral-containers.md

```diff
@@ -76,3 +76,195 @@ you can view processes in other containers.
+
+
+## Digging Deeper
+
+Now that we have the new feature we can start a ephemeral container in any Pod we like  
+
+kubectl run ephemeral-demo --image=k8s.gcr.io/pause:3.1 --restart=Never
+kubectl debug -it ephemeral-demo --image=busybox:1.28
+
+We get a shell and life is now much simpler, but wait a minute. This post is not about how to use ephemeral containers, there are enough of those already, but rather the security implications of enabling ephemeral containers. Let's have a look at the YAML for the Pod that we created the ephemeral container in.
+
+apiVersion: v1
+kind: Pod
+metadata:
+  name: ephemeral-demo
+  spec:
+    ...
+      ephemeralContainers:
+        - name: debugger-r59b7
+            image: busybox:1.28
+                imagePullPolicy: IfNotPresent
+                    stdin: true
+                        terminationMessagePath: /dev/termination-log
+                            terminationMessagePolicy: File
+                                tty: true
+                            
+Interesting, there is a new field called ephemeralContainers in the Pod definition. This new field contains a list of containers similar to initContainers and containers. It is not identical as there are certain options which are not available, refer to the API documentation for more information. It does however allow configuration of the container security context, which could in theory allow a bad actor to escalate the container's privileges. This should not affect those of us who use a policy enforcement tool right? The answer is yes and no depending on the tool and version that is being used. It also depends on if you are using policies from the project's library or policies developed in house.
+
+
+## OPA Gatekeeper
+
+OPA Gatekeeper does not require any code changes as all of its policies are written in rego. It's sub project Gateekper Library does however have to be updated. The library contains an implementation of the common Pod Security Policies. This includes policies like not allowing containers in privileged mode. The issue with the all of the policies is that they currently only check containers specified in initContainers and containers, analyze the following rego as an example.
+
+The good news is that this is a pretty easy fix, the bad news is that it requires end users to update the policies pulled from the library.
+
+## Kyverno
+
+Kyverno seems to have resolved the issues faster. Compared to OPA Gatekeeper however it did require a small code change which means that version 1.5.3 or later is needed to write policies for ephemeral containers. They have also updated their policy library to include checking ephemeral containers. Kyverno has done a great job solving these issues quickly. It does still require end users to update however.
+
```

Comment:
No. We don't talk about community solutions that are out of the kubernetes project.
There are simply too many of them, we cannot afford to cover or to track.